samples/discoverNatBehavior.c

```diff
@@ -0,0 +1,96 @@
+#include <com/amazonaws/kinesis/video/webrtcclient/Include.h>
+
+#define NETWORK_INTERFACE_NAME_PARAM                "-i"
+#define STUN_HOSTNAME_PARAM                         "-s"
+#define DEFAULT_STUN_HOST                           "stun:stun.sipgate.net:3478"
+
+BOOL filterFunc(UINT64 data, PCHAR name)
+{
+    CHAR *desiredInterface = (CHAR *) data;
+    if (desiredInterface == NULL || STRCMP(name, desiredInterface) == 0) {
+        return TRUE;
+    }
+    return FALSE;
+}
+
+INT32 main(INT32 argc, CHAR **argv)
+{
+    PCHAR interfaceName = NULL, stunHostname = NULL;
+    printf("Usage: ./discoverNatBehavior -i network-interface-name -s stun-hostname\n");
+    NAT_BEHAVIOR mappingBehavior = NAT_BEHAVIOR_NONE, filteringBehavior = NAT_BEHAVIOR_NONE;
+    INT32 i;
+
+    for (i = 1; i < argc; ++i) {
+        if (STRCMP(argv[i], NETWORK_INTERFACE_NAME_PARAM) == 0) {
+            interfaceName = argv[++i];
+            i++;
+        } else if (STRCMP(argv[i], STUN_HOSTNAME_PARAM) == 0) {
+            stunHostname = argv[++i];
+            i++;
+        } else {
+            printf("Unknown param %s", argv[i]);
+        }
+    }
+
+    if (stunHostname == NULL) {
+        stunHostname = DEFAULT_STUN_HOST;
+    }
+
+    printf("Using stun host: %s, local network interface %s.\n", stunHostname, interfaceName);
+
+    initKvsWebRtc();
+    SET_LOGGER_LOG_LEVEL(LOG_LEVEL_DEBUG);
+
+    if (discoverNatBehavior(stunHostname, &mappingBehavior, &filteringBehavior, filterFunc, (UINT64) interfaceName)) {
```

Comment:
This returns STATUS. Check against STATUS_FAILED and print out the error?

Reply:
fixed.

Reply:
fixed